Feature comments

Kazantsevyury opened the pull request:
feat: Implement additional functionality

### Stage 3: Additional Functionality
additional functionality:
- `comments` - comments on events

Kazantsevyury committed:
feat(comments): add functionality for event comments

Implemented the functionality to allow users to add comments to events.

Kazantsevyury committed:
feat(comments): add functionality for event comments

Implemented the functionality to allow users to add comments to events.

Kazantsevyury committed:
refactor

Kazantsevyury committed:
refactor

Kazantsevyury committed:
refactor: Adding documentation. Logging refactoring

Kazantsevyury committed:
refactor: Adding documentation. Logging refactoring

Kazantsevyury committed:
docs: add documentation for better readability

Kazantsevyury committed:
docs: add documentation for better readability

Kazantsevyury committed:
docs: add documentation for better readability

Kazantsevyury committed:
docs: add documentation for better readability

Kazantsevyury committed:
refactor(EventRepository, CommentRepository): improve query readability and maintainability

Kazantsevyury committed:
docs: add documentation for better readability

Kazantsevyury committed:
feat(stat-client): optimize StatClient module and classes

Kazantsevyury committed:
Checkstyle

Kazantsevyury committed:
feat(stat-server): optimize module and classes.Adding documentation. Logging refactoring

Kazantsevyury committed:
feat(pom): refactor and optimize all pom files

Kazantsevyury committed:
add postman tests

Kazantsevyury committed:
add README.md

PolnySkvorets commented on ewm-service/src/main/java/ru/practicum/yandex/events/dto/EventFullDto.java:
```diff
@@ -53,4 +54,6 @@ public class EventFullDto {
 
     private LocationDto location;
+
+    List<CommentDto> comments;
```
```suggestion
    private List<CommentDto> comments;
```

PolnySkvorets commented on ewm-service/src/main/java/ru/practicum/yandex/events/repository/EventSpecification.java:
```diff
@@ -1,12 +1,14 @@
 import java.time.LocalDateTime;
 import java.util.List;
 
+@UtilityClass
```
не стоит использовать экспериментальные аннотации

PolnySkvorets reviewed (commented):
Оставил пару замечаний

Kazantsevyury committed:
refactor(events-repository): remove @UtilityClass annotation from EventSpecification, add  documentation

Refactored the EventSpecification class to remove the experimental @UtilityClass annotation.
Added documentation for each method in the EventSpecification class to explain its purpose and parameters.
Improved null checks by using Specification.where(null) for better integration with Spring Data JPA.

Kazantsevyury committed:
docs(repositories): add documentation and improve formatting

- Added Javadoc comments for methods and interfaces in EventRepository and CompilationRepository to explain their purpose and parameters.
- Improved the readability of JPQL queries by enhancing their formatting.
- Ensured consistency in code style across both repositories without changing functionality.
- Replaced null returns in EventSpecification with Specification.where(null) for better integration with Spring Data JPA.

PolnySkvorets reviewed (approved):
Принято
